ts/components/ui/Markdown/rules/link.ts

```diff
@@ -13,20 +13,16 @@ type ScreenNameToRouteMap = {
 };
 
 // Regex to match uri like `internal://screen/profile`
-const INTERNAL_TARGET_REGEX = /^internal:\/\/screen\/([\w_-]+)$/;
+const INTERNAL_TARGET_REGEX_PATTERN = /^internal:\/\/screen\/([\w_-]+)$/;
 
 // Here we put all the allowed screen name with the mapping to the react-navigation route
 const SCREEN_NAME_TO_ROUTE_MAP: ScreenNameToRouteMap = {
   profile: ROUTES.PROFILE_MAIN
 };
 
 function getInternalRoute(target: string): Option<string> {
-  const match = INTERNAL_TARGET_REGEX.exec(target);
+  const match = new RegExp(INTERNAL_TARGET_REGEX_PATTERN).exec(target);
```

Comment:
I don't know if it makes sense to create a regexp for something so simple, what about:

```ts
const parsed = target.split("internal://screen/");

if(parsed.length === 2 && parsed[0] === "") {
  const screenName = parsed[1];
  // ...
}
```